qiita_pet/handlers/compute.py

```diff
@@ -28,6 +30,49 @@ def get(self, job_id):
         self.redirect('/')
 
 
+class CreateRawData(BaseHandler):
+    @authenticated
+    def post(self):
+        pt_id = self.get_argument('prep_template_id')
+        raw_data_filetype = self.get_argument('filetype')
+        barcodes_str = self.get_argument('barcodes')
+        forward_reads_str = self.get_argument('forward')
+        sff_str = self.get_argument('sff')
+        fasta_str = self.get_argument('fasta')
+        qual_str = self.get_argument('qual')
+        reverse_reads_str = self.get_argument('reverse')
+
+        pt = PrepTemplate(pt_id)
+        study_id = pt.study_id
+
+        def _split(x):
+            return x.split(',') if x else []
+        filepaths, fps = [], []
+        fps.append((_split(barcodes_str), 'raw_barcodes'))
+        fps.append((_split(fasta_str), 'raw_fasta'))
+        fps.append((_split(qual_str), 'raw_qual'))
+        fps.append((_split(forward_reads_str), 'raw_forward_seqs'))
+        fps.append((_split(reverse_reads_str), 'raw_reverse_seqs'))
+        fps.append((_split(sff_str), 'raw_sff'))
+
+        for _, f in get_mountpoint("uploads", retrieve_all=True):
```

Comment:
Would you mind adding a comment describing this block, it took me a while to understand it :eyeglasses: